letsencrypt/client/tests/standalone_authenticator_test.py

```diff
@@ -0,0 +1,564 @@
+                         ["dvsni"])
+
+
+class SNICallbackTest(unittest.TestCase):
+    """Tests for sni_callback() method."""
+    def setUp(self):
+        from letsencrypt.client.standalone_authenticator import \
+            StandaloneAuthenticator
+        from letsencrypt.client.challenge_util import dvsni_gen_cert
+        from letsencrypt.client import le_util
+        import OpenSSL.crypto
+        from OpenSSL.crypto import FILETYPE_PEM
+        self.authenticator = StandaloneAuthenticator()
+        name, r_b64 = "example.com", le_util.jose_b64encode("x" * 32)
+        test_key = pkg_resources.resource_string(__name__,
+                                                 'testdata/rsa256_key.pem')
+        nonce, key = "abcdef", le_util.Key("foo", test_key)
+        self.cert = dvsni_gen_cert(name, r_b64, nonce, key)[0]
+        private_key = OpenSSL.crypto.load_privatekey(FILETYPE_PEM, key.pem)
+        self.authenticator.private_key = private_key
+        self.authenticator.tasks = {"abcdef.acme.invalid": self.cert}
+        self.authenticator.child_pid = 12345
+
+    def test_real_servername(self):
+        import OpenSSL.SSL
+        connection = mock.MagicMock()
+        connection.get_servername.return_value = "abcdef.acme.invalid"
+        self.authenticator.sni_callback(connection)
+        self.assertEqual(connection.set_context.call_count, 1)
+        called_ctx = connection.set_context.call_args[0][0]
+        self.assertTrue(isinstance(called_ctx, OpenSSL.SSL.Context))
+
+    def test_fake_servername(self):
+        """Test the behavior of the SNI callback when an unexpected SNI
+        name is received.  (Currently the expected behavior in this case
+        is to return the "first" certificate with which the listener
+        was configured, although they are stored in an unordered data
+        structure so this might not be the one that was first in the
+        challenge list passed to the perform method.  In the future, this
+        might result in dropping the connection instead.)"""
+        import OpenSSL.SSL
+        connection = mock.MagicMock()
+        connection.get_servername.return_value = "example.com"
+        self.authenticator.sni_callback(connection)
+        self.assertEqual(connection.set_context.call_count, 1)
+        called_ctx = connection.set_context.call_args[0][0]
+        self.assertTrue(isinstance(called_ctx, OpenSSL.SSL.Context))
+
+class ClientSignalHandlerTest(unittest.TestCase):
+    """Tests for client_signal_handler() method."""
+    def setUp(self):
+        from letsencrypt.client.standalone_authenticator import \
+            StandaloneAuthenticator
+        self.authenticator = StandaloneAuthenticator()
+        self.authenticator.tasks = {"foononce.acme.invalid": "stuff"}
+        self.authenticator.child_pid = 12345
+
+    def test_client_signal_handler(self):
+        import signal
+        self.assertEqual(self.authenticator.subproc_state, None)
```

Comment:
Please don't do `x == None`! `self.assertTrue(self.authenticator.subproc_state is None)`